tests/utilities/test_deprecation.py

```diff
@@ -0,0 +1,37 @@
+import pytest
+
+from pytorch_lightning.utilities.deprecation import _deprecated
+
+
+def my_sum(a, b=3):
+    return a + b
+
+
+@_deprecated(target=my_sum, ver_deprecate="0.1", ver_remove="0.5")
+def dep_sum(a, b):
+    pass
+
+
+@_deprecated(target=my_sum, ver_deprecate="0.1", ver_remove="0.5")
+def dep2_sum(a, b):
+    pass
+
+
+def test_deprecated_func():
+    with pytest.deprecated_call(
+        match='This `dep_sum` was deprecated since v0.1 in favor of `tests.utilities.test_deprecation.my_sum`.'
+        ' It will be removed in v0.5.'
+    ):
+        assert dep_sum(2, b=5) == 7
+
+    # check that the warning is raised only once per function
+    with pytest.warns(None) as record:
```

Comment:
@Borda You could use this https://github.com/PyTorchLightning/pytorch-lightning/blob/master/tests/helpers/utils.py#L121

```python
with no_warning_call(DeprecationWarning):
	...
```